core/src/main/java/com/github/mizool/core/configuration/ConfigurationModule.java

```diff
@@ -0,0 +1,17 @@
+package com.github.mizool.core.configuration;
+
+import org.kohsuke.MetaInfServices;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Module;
+
+@MetaInfServices(Module.class)
+public class ConfigurationModule extends AbstractModule
```

Comment:
This is the wrong place for this module. Loading environment variables instead of system properties came up with lambdas. Putting it here as a service provider for the `com.google.inject.Module` interface hardwires guice to to environment variables and thus lambdas (at least for now).

The opposite goes for the `@Vetoed` annotation on the `EnvironmentVariableConfigurationImpl`.

I wonder if `ConfigurationModule` and `EnvironmentVariableConfigurationImpl` belong into `technology/lambda`. However, I don't see where to put `SystemPropertyConfigurationImpl` so it doesn't collide. `technology/not-lambda` sounds cumbersome.

We could put `ConfigurationModule` and `EnvironmentVariableConfigurationImpl` into a lambda technology and have `EnvironmentVariableConfigurationImpl` extend `SystemPropertyConfigurationImpl` with a `@Specializes` annotation. But that feels wrong as well.

Perhaps @bannmann has an idea when he reviews this?

Reply:
My suggestion would be to move this to technology-lambda once we actually create that technology when the lambda stuff from genius has matured.

Reply:
To be more precise: The vetoed implementation offers an alternative that is inactive by default. Any consumer is free to activate it in any way deemed proper. Once the module that activates it moves to the lambda technology, the concept becomes clean again.

Reply:
Let's solve this dilemma as follows:

- `SystemPropertyConfigurationImpl` is kept as is. CDI based applications will use it by default.
- To avoid a general "guice means environment variable config" assumption in mizool-core, `ConfigurationModule` and `EnvironmentVariableConfigurationImpl` remain in the downstream project (Genius) for now. They keep their respective `@MetaInfServices` and `@Vetoed` annotations. 
    - When we move the lambda stuff from downstream to mizool, `ConfigurationModule` ends up in `technology-lambda`. At that point, we can move `EnvironmentVariableConfigurationImpl` there as well or to core, but neither option should cause problems for the CDI world.